molgenis/molgenis-emx2
# feat: submission feature (alpha)

This feature enables to
* create a submission by 'cloning' a part of a schema + contents into a 'submission'.  
* then user can edit this clone.
* finally, admin can then 'merge' the submission, processing 'deletions' and 'saves' into the original schema (aka 'targetSchema).

Currently only configured for 'data catalogue' and then 'cohorts'.

To test:
- create a schema using datacatalogue template
- in this schema browse to /myschema/submit

Here you can play around, will look something like this:

<img width="1151" alt="image" src="https://user-images.githubusercontent.com/120060/186273630-c97d45e6-50f1-45fd-a22d-4d5c2dd7bfeb.png">

In above dialogue you selected 'TestCohort' so therefore some data will have been copied related to TestCohort

<img width="1140" alt="image" src="https://user-images.githubusercontent.com/120060/186273712-e8ca6b32-84ce-46a9-a40c-383a2637413b.png">

You can see listing of submissions into this schema

<img width="1180" alt="image" src="https://user-images.githubusercontent.com/120060/186273779-ece3b3c4-eacb-489d-8270-a12879212aee.png">

When pushing merge, results will be merged

<img width="1164" alt="image" src="https://user-images.githubusercontent.com/120060/186275370-07cf5b34-100d-4f61-9ba3-6104b3eebf8b.png">
